### PR TITLE
Ignore files matching `.gitignore` patterns

### PR DESCRIPTION
**Pull request by retzion**

### CODE REVIEW

**Comment by retzion** [comment]
This gitignore parser turned out to be a dud. It doesn't even handle `*` wildcards yet and he says he's not developing the package any more.
https://github.com/mherrmann/gitignore_parser

I'm not finding anything else that works, so I'm building my own rudimentary parser. I've tested on patterns such as:
```
directory_name/
*.ext
```

I am continuing to support patterns such as:
```
*.directory_name/
*.ext
*.py[cod]
```
See: https://git-scm.com/docs/gitignore#_pattern_format

**Comment by retzion** [comment]
We will write a PIP package for this and publish it as open-source, unlicensed. I'll clean up this branch and PR after I publish to PyPi.

**Comment by retzion** [comment]
Some valuable pieces in here. Even though the gitignore plan has changed, I'll stay with this PR.

**Comment by retzion** [comment]
todo: Test the Advanced Use Cases listed at https://labs.consol.de/development/git/2017/02/22/gitignore.html

File: bugcatcher/ftl.py [hunk 1]
```diff
@@ -23,7 +24,7 @@
 # from Login import current_login
 
 args = None
-default_extensions = ['.sol', '.json', '.txt', '.py']
+default_extensions = []
```
Comment (tqman) [inline bugcatcher/ftl.py:27]: Do we want to have the default be none? What about the case where the project isn't in git?

**Review by tqman — APPROVED** [review]

**Pull request merged**